scarlet/display.py

```diff
@@ -1,224 +1,268 @@
+            filter_weights /= 1.667
+        if B == 6:
+            filter_weights[0,5] = 1
+            filter_weights[0,4] = 0.667
+            filter_weights[0,3] = 0.333
+            filter_weights[1,4] = 0.333
+            filter_weights[1,3] = 0.667
+            filter_weights[1,2] = 0.667
+            filter_weights[1,1] = 0.333
+            filter_weights[2,2] = 0.333
+            filter_weights[2,1] = 0.667
+            filter_weights[2,0] = 1
+            filter_weights /= 2
     else:
-        rgb = norm(img)
+        assert filter_weights.shape == (3, len(img))
+
+    # map bands onto RGB channels
+    _,ny,nx = img_.shape
+    rgb = np.dot(filter_weights, img_.reshape(B,-1)).reshape(3,ny,nx)
+
     if hasattr(rgb, "mask"):
         rgb = rgb.filled(fill_value)
-    # The result of calling Asinh is an array mapped to [0,1]
-    # We need to map this to a uint8 in the range [0,255]
-    rgb *= 255
-    rgb[rgb < 0] = 0
-    rgb[rgb > 255] = 255
-    rgb = rgb.astype(np.uint8)
-    # Now we move the SED to the outer axis and use only the selected colors
-    rgb = np.transpose(rgb[filter_indices], axes=(1, 2, 0))
+
     return rgb
 
+def channel_to_rgb(channels, norm):
+    """Convert RGB channel values to [0..255]
 
-def zscale(img, contrast=0.25, samples=500, fraction=.75):
-    """Calculate minimum and maximum pixel values based on the image
+    If normalized values are outside of the range [0..255], they will be
+    truncated such as to preserve the corresponding color.
+
+    Parameters
+    ----------
+    channels: array_like
+        Converted RGB channel values with dimension (3, height, width)
+    norm: `scarlet.display.Norm`
+        Norm to use for mapping in the allowed range [0..255]
+
+    Returns
+    -------
+    rgb: numpy array with dimensions (3, height, width) and dtype uint8
     """
-    if img.size > samples:
-        sorted_img = np.sort(np.random.choice(img.reshape(-1), size=samples))
-    else:
-        sorted_img = np.sort(img.reshape(-1))
+    rgb = norm(channels)
 
-    n_pix = len(sorted_img)
-    center = n_pix/2
-    median = sorted_img[int(center)]
-    idx = np.arange(n_pix)
+    # truncate at 0 and 1
+    rgb[rgb < 0] = 0 # looks better but is inaccurate because colors are not treated the same
+    # normalizing by a uniform maximum is critical for the color of saturated regions
+    rgb_max = rgb.max(axis=0)
+    rgb[:, rgb_max > 1] /= rgb_max[rgb_max > 1]
 
-    radius = 1-fraction/2
-    sample_slice = slice(int(np.floor((0.5-radius)*n_pix)), int(np.ceil((0.5+radius)*n_pix)))
-    result = np.polyfit(idx[sample_slice], sorted_img[sample_slice], 1)
-    slope, intercept = result
+    maxRGB = 255
+    return np.dstack((rgb * maxRGB)).astype(np.uint8)
 
-    z1 = max(median - center * (slope/contrast), np.min(img))
-    z2 = min(median + (n_pix-center-1) * (slope/contrast), np.max(img))
-    return z1, z2
+def rgb(img, filter_weights=None, fill_value=0, norm=None):
```

Comment:
I think this should be called `img_to_rgb`. You use the parameter `rgb` in some of your other functions and this naming convention makes it clear what this function does.